service/src/test/java/uk/gov/hmcts/reform/fpl/service/CreateC21OrderServiceTest.java

```diff
@@ -0,0 +1,232 @@
+            .legalAdvisorName("Peter Parker")
+            .build());
+
+        assertThat(returnedOrder.getOrderTitle()).isEqualTo("Example Title");
+        assertThat(returnedOrder.getDocument()).isEqualTo(DocumentReference.builder().build());
+        assertThat(returnedOrder.getOrderDetails()).isEqualTo("Some details");
+        assertThat(returnedOrder.getOrderDate()).isNotNull();
+        assertThat(returnedOrder.getJudgeAndLegalAdvisor()).isEqualTo(JudgeAndLegalAdvisor.builder()
+            .judgeTitle(HER_HONOUR_JUDGE)
+            .judgeLastName("Judy")
+            .legalAdvisorName("Peter Parker")
+            .build());
+    }
+
+    @Test
+    void shouldCreateExpectedMapWhenGivenPopulatedCaseData() {
+        LocalDate localDate = LocalDate.now();
+        String date = localDate.format(DateTimeFormatter.ofLocalizedDate(FormatStyle.LONG));
+
+        CaseData caseData = populatedCaseData(localDate);
+
+        Map<String, Object> expectedMap = expectedData(date);
+
+        Map<String, Object> templateData = service.getC21OrderTemplateData(caseData);
+        assertThat(templateData).isEqualTo(expectedMap);
+    }
+
+    @SuppressWarnings("unchecked")
+    private Map<String, Object> expectedData(String date) {
+        ImmutableMap.Builder expectedMap = ImmutableMap.<String, Object>builder();
+
+        expectedMap.put("familyManCaseNumber", "123")
+            .put("courtName", "Example Court")
+            .put("orderTitle", "Example Title")
+            .put("orderDetails", "Example details")
+            .put("todaysDate", date)
+            .put("judgeTitleAndName", "Her Honour Judge Judy")
+            .put("legalAdvisorName", "Peter Parker")
+            .put("children", ImmutableList.of(
+                ImmutableMap.of(
+                    "name", "Timmy Jones",
+                    "gender", "Boy",
+                    "dateOfBirth", date)));
+
+        return expectedMap.build();
+    }
+
+    private CaseData populatedCaseData(LocalDate localDate) {
+        return CaseData.builder()
+            .familyManCaseNumber("123")
+            .caseLocalAuthority("example")
+            .c21Order(C21Order.builder()
+                .orderTitle("Example Title")
+                .orderDetails("Example details")
+                .build())
+            .judgeAndLegalAdvisor(JudgeAndLegalAdvisor.builder()
+                .judgeTitle(HER_HONOUR_JUDGE)
+                .judgeLastName("Judy")
+                .legalAdvisorName("Peter Parker")
+                .build())
+            .children1(ImmutableList.of(Element.<Child>builder()
+                .value(Child.builder()
+                    .party(ChildParty.builder()
+                        .firstName("Timmy")
+                        .lastName("Jones")
+                        .gender("Boy")
+                        .dateOfBirth(localDate)
+                        .build())
+                    .build())
+                .build()))
+            .build();
+    }
```

Comment:
could we use callbackRequest() here instead?

Reply:
map that to caseData...

Reply:
problem with dateOfBirth if we do this - decided to keep as is